plone/app/contenttypes/migration/utils.py

```diff
@@ -253,6 +254,8 @@ def migrate_portlets(src_obj, dst_obj):
 
     # copy all portlet assignments (visibilty is stored as annotation
     # on the assignments and gets copied here too)
+    if not ILocalPortletAssignable.providedBy(src_obj):
+        return
```

Comment:
Thanks. Please also test the dst_obj for that. I added that after my initial comment so It was probably not in the email.